src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,184 @@
+from itertools import permutations
+
+import pint
+from pint import Unit
+
+unit = pint.UnitRegistry(autoconvert_offset_to_baseunit=True)
+unit.default_system = "SI"
+
+# Unit definitions
+unit.define("1.e15-particles/kg = 1 kg^-1 = 1^15particles kg^-1")
+unit.define("1.e15-particles m^-3 s^-1 = 1 m^-3 s^-1 = 1^15particles m^-3 s^-1")
+unit.define("ton_force = 1 UK_force_ton = tonf")
+
+
+quantity = unit.Quantity
+
+
+restricted_units = ["Hz", "hertz", "rad/s", "radian/s", "rpm", "rps", "cps"]
+restricted_conversions = list((permutations(restricted_units, 2)))
+
+
+std_prefixes = [
+    "y",
+    "z",
+    "a",
+    "f",
+    "p",
+    "n",
+    "u",
+    "c",
+    "d",
+    "da",
+    "h",
+    "m",
+    "k",
+    "M",
+    "G",
+    "T",
+    "P",
+    "E",
+    "Z",
+    "Y",
+    "",
+]
+
+
+restricted_unit_expansion = {
+    "Hz": [prefix + "Hz" for prefix in std_prefixes],
+    "hertz": [prefix + "hertz" for prefix in std_prefixes],
+    "rad/s": ["radian/s", "rad/s"],
+    "radian/s": ["rad/s", "radian/s"],
+    "rpm": ["revolutions_per_minute", "rpm"],
+    "rps": ["revolutions_per_second", "rps"],
+    "cps": ["counts_per_second", "cps"],
+}
+
+
+def build_restricted_conversions(conversions, unit_expansion):
+
+    """This function generates required final restricted mappings."""
+
+    keys = set([i[0] for i in conversions])
+    restricted_units_dict = {unit[0]: [] for unit in conversions}
+
+    for key in keys:
+        temp_list = [unit_expansion[i] for i in keys.difference(set([key]))]
+        for temp in temp_list:
+            restricted_units_dict[key] += temp
+    for key in keys:
+        restricted_units_dict[key] = list(set(restricted_units_dict[key]))
+    return restricted_units_dict
+
+
+restricted_units = build_restricted_conversions(
+    restricted_conversions, restricted_unit_expansion
+)
+
+
+class Quantity:
+
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    All the instances of this class are converted to base SI units
+    system. Any conversion between "Hz", "hertz", "rad/s", "radian/s",
+    "rpm", "rps", "cps" is restricted.
+    """
+
+    def __init__(self, real_value, units_string):
+        self.value = real_value
+        self.unit = units_string
+        self._quantity = quantity(self.value, self.unit)
```

Comment:
Does `self._quantity` also cache the `real_value`?  If so, do you need the redundant copy?

Reply:
Yes it does. We are using  ```self.value``` and ```self.unit``` in ```__float__```,  ```__str__``` and ```__repr__``` methods. 
also if we directly use ```real_value``` to instantiate ```self._quanity``` it modifies it's precision according to pint's default format and we can not represent our original Quantity instance as per the discussion in proposal. Hence we need the redundant copy as well.

Reply:
Yeah, I guess I just wondered if you could replace the cached copy with
```
def value(self):
    return self._quantity.magnitude
```
but if the `quantity` is modifying the passed value then maybe it's fine.